clients/python/girder_client/__init__.py

```diff
@@ -1012,6 +1012,42 @@ def downloadFolderRecursive(self, folderId, dest, sync=False):
             if len(items) < DEFAULT_PAGE_LIMIT:
                 break
 
+    def downloadResource(self, resourceId, dest, resourceType='folder', sync=False):
+        """
+        Download a folder recursively from Girder into a local directory.
+
+        :param resourceId: ID or path of the resource to download.
```

Comment:
Well, ID is short for identifier, so I'm not sure about that nomenclature. `idOrPath` seems more descriptive, what do you think?